firstvoices/backend/tests/test_apis/test_characters_api.py

```diff
@@ -142,3 +143,176 @@ def test_detail_related_entries(self):
+        site = factories.SiteFactory.create(visibility=Visibility.PUBLIC)
+        speaker = factories.PersonFactory.create(site=site)
+        audio = factories.AudioFactory.create(site=site)
+        image = factories.ImageFactory.create(site=site)
+        video = factories.VideoFactory.create(site=site)
+        factories.AudioSpeakerFactory.create(speaker=speaker, audio=audio, site=site)
+
+        instance = self.create_instance_with_media(
+            site=site,
+            visibility=Visibility.PUBLIC,
+            related_audio=(audio,),
+            related_images=(image,),
+            related_videos=(video,),
+        )
+
+        new_audio = factories.AudioFactory.create(site=site)
+        factories.AudioSpeakerFactory.create(
+            speaker=speaker, audio=new_audio, site=site
+        )
+        new_image = factories.ImageFactory.create(site=site)
+        new_video = factories.VideoFactory.create(site=site)
+
+        req_body = {
+            "related_audio": [str(new_audio.id)],
+            "related_images": [str(new_image.id)],
+            "related_videos": [str(new_video.id)],
+        }
+
+        user = factories.get_app_admin(role=AppRole.SUPERADMIN)
+        self.client.force_authenticate(user=user)
+
+        response = self.client.put(
+            self.get_detail_endpoint(key=instance.id, site_slug=site.slug),
+            format="json",
+            data=req_body,
+        )
+        assert response.status_code == 200
+        response_data = json.loads(response.content)
+
+        assert len(response_data["relatedAudio"]) == 1
+        assert response_data["relatedAudio"][0] == self.get_expected_audio_data(
+            new_audio, speaker
+        )
+        assert len(response_data["relatedImages"]) == 1
+        assert response_data["relatedImages"][0] == self.get_expected_image_data(
+            new_image
+        )
+        assert len(response_data["relatedVideos"]) == 1
+        assert response_data["relatedVideos"][0] == self.get_expected_video_data(
+            new_video
+        )
+
+        assert (
+            Character.objects.get(id=instance.id).related_audio.all().first().id
+            == new_audio.id
+        )
+        assert (
+            Character.objects.get(id=instance.id).related_images.all().first().id
+            == new_image.id
+        )
+        assert (
+            Character.objects.get(id=instance.id).related_videos.all().first().id
+            == new_video.id
+        )
+
+    @pytest.mark.django_db
+    def test_update_character_fields(self):
+        site = factories.SiteFactory.create(visibility=Visibility.PUBLIC)
+        character = factories.CharacterFactory.create(site=site)
+        dictionary_entry = factories.DictionaryEntryFactory.create(site=site)
+
+        req_body = {
+            "note": self.CHARACTER_NOTE,
+            "related_dictionary_entries": [str(dictionary_entry.id)],
+        }
+
+        user = factories.get_app_admin(role=AppRole.SUPERADMIN)
```

Comment:
Perhaps it would be better to check that these tests work for the language admin role? (or both)